test/test_auth_endpoint.py

```diff
@@ -1,36 +1,46 @@
+from config import test_config
 import pytest
 import requests
 import base64
 import re
 import logging
 import html
 
-# Pragmatic
-# Search for IRODS_HTTP_API_BASE_URL
-# and IRODS_CLIENT_VERSION
-# TOOD: Consider turning this into a fixture
-AUTH_URL = 'http://127.0.0.1:9000/irods-http-api/0.2.0/authenticate'
+skip_protected_resource = pytest.mark.skipif(
+    test_config['openid_connect']['mode'] == 'protected_resource', reason='Not supported in "protected_resource" mode.')
 
-@pytest.mark.parametrize("username, password, expected_result", [('bob', 'bob', requests.codes.ok),
-                                                                 ('', '', requests.codes.unauthorized),
+skip_client = pytest.mark.skipif(
+    test_config['openid_connect']['mode'] == 'client', reason='Not supported in "client" mode.')
```

Comment:
What happens if the config options aren't defined?

Reply:
Resolved with schema.